WNPRC_Purchasing/src/org/labkey/wnprc_purchasing/WNPRC_PurchasingController.java

```diff
@@ -112,19 +152,468 @@ public void addNavTrail(NavTree root) { }
         @Override
         public Object execute(RequestForm requestForm, BindException errors) throws Exception
         {
+            //get old line items to compare the difference with new incoming changes
+            List<LineItem> oldLineItems = null;
+            if (requestForm.getRowId() != null)
+            {
+                TableInfo tableInfo = QueryService.get().getUserSchema(getUser(), getContainer(), "ehr_purchasing").getTable("lineItems", null);
+                Filter filter = new SimpleFilter(FieldKey.fromString("requestRowId"), requestForm.getRowId());
+                TableSelector tableSelector = new TableSelector(tableInfo, filter, null);
+                oldLineItems = tableSelector.getArrayList(LineItem.class);
+            }
+
+            //get old status to compare the difference with new status change
+            String oldStatusVal = "";
+            if (requestForm.getRowId() != null)
+            {
+                TableInfo tableInfo = QueryService.get().getUserSchema(getUser(), getContainer(), "ehr_purchasing").getTable("purchasingRequests", null);
+                Filter filter = new SimpleFilter(FieldKey.fromString("rowId"), requestForm.getRowId());
+                Set<FieldKey> columns = new HashSet<>();
+                columns.add(FieldKey.fromString("rowId"));
+                columns.add(FieldKey.fromString("qcState/Label"));
+                final Map<FieldKey, ColumnInfo> colMap = QueryService.get().getColumns(tableInfo, columns);
+                TableSelector tableSelector = new TableSelector(tableInfo, colMap.values(), filter, null);
+                oldStatusVal = String.valueOf(tableSelector.getMap().get("qcstate_fs_label"));
```

Comment:
oh ok, ive seen this in a few places, so thought it was ok. didn't realize getAlias() would do the trick, now changed

Reply:
I'm less nervous about it when it's a column that's directly on a table instead of a lookup, but still best to let Query choose whatever it wants during SQL generation